frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/dental-insurance.tsx

```diff
@@ -0,0 +1,215 @@
+import { getFixedT } from '~/utils/locale-utils.server';
+import { getLogger } from '~/utils/logging.server';
+import { mergeMeta } from '~/utils/meta-utils';
+import { getPathById } from '~/utils/route-utils';
+import { getTitleMetaTags } from '~/utils/seo-utils';
+import { cn } from '~/utils/tw-utils';
+
+export const handle = {
+  i18nNamespaces: getTypedI18nNamespaces('apply-adult-child', 'apply', 'gcweb'),
+  pageIdentifier: pageIds.public.apply.adultChild.dentalInsurance,
+  pageTitleI18nKey: 'apply-adult-child:children.dental-insurance.title',
+};
+
+export const meta: MetaFunction<typeof loader> = mergeMeta(({ data }) => {
+  return data ? getTitleMetaTags(data.meta.title) : [];
+});
+
+export async function loader({ context: { session }, params, request }: LoaderFunctionArgs) {
+  const state = loadApplyAdultChildState({ params, request, session });
+  const t = await getFixedT(request, handle.i18nNamespaces);
+  const child = state.children.find((child) => child.id === params.childId);
+  const childName = child ? child.information?.firstName : '<Child 1 name>';
+
+  const csrfToken = String(session.get('csrfToken'));
+  const meta = { title: t('gcweb:meta.title.template', { title: t('apply-adult-child:children.dental-insurance.title') }) };
+
+  return json({ id: state, csrfToken, meta, defaultState: state.dentalInsurance, childName, editMode: state.editMode });
+}
+
+export async function action({ context: { session }, params, request }: ActionFunctionArgs) {
+  const log = getLogger('apply/adult-child/dental-insurance');
+
+  const state = loadApplyAdultChildState({ params, request, session });
+  const t = await getFixedT(request, handle.i18nNamespaces);
+
+  // state validation schema
+  const dentalInsuranceSchema = z.boolean({ errorMap: () => ({ message: t('apply-adult-child:children.dental-insurance.error-message.dental-insurance-required') }) });
+
+  const formData = await request.formData();
+  const expectedCsrfToken = String(session.get('csrfToken'));
+  const submittedCsrfToken = String(formData.get('_csrf'));
+
+  if (expectedCsrfToken !== submittedCsrfToken) {
+    log.warn('Invalid CSRF token detected; expected: [%s], submitted: [%s]', expectedCsrfToken, submittedCsrfToken);
+    throw new Response('Invalid CSRF token', { status: 400 });
+  }
+
+  const dentalInsurance = formData.get('dentalInsurance') ? formData.get('dentalInsurance') === 'yes' : undefined;
+  const parsedDataResult = dentalInsuranceSchema.safeParse(dentalInsurance);
+
+  if (!parsedDataResult.success) {
+    return json({ errors: parsedDataResult.error.format()._errors });
+  }
+
+  saveApplyState({ params, session, state: { dentalInsurance: parsedDataResult.data } });
+
+  if (state.editMode) {
+    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult-child/review-information', params));
+  }
+
+  return redirect(getPathById('$lang+/_public+/apply+/$id+/adult-child/federal-provincial-territorial-benefits', params));
```

Comment:
Redirect to children HUB for now.

```suggestion
  return redirect(getPathById('$lang+/_public+/apply+/$id+/adult-child/children/index', params));
```